docs/deploying.md

```diff
@@ -90,6 +96,48 @@ kustomize build $BMOPATH/ironic-deployment/default | kubectl apply -f-
 
 where $BMOPATH points to the baremetal-operator path.
 
+### Deploying with TLS
+
+For this, you need to first copy your Ironic CA Certificate, then build the
+kustomization for baremetal operator:
+
+```sh
+   cp <path-to-ca-certificate> $BMOPATH/deploy/tls/ca.crt
+   ./tools/bin/kustomize build $BMOPATH/deploy/tls | kubectl apply -f -
+```
+
+Then you need to copy all your certificates and build the kustomization for
+ironic:
+
+```sh
+   cp <path-to-ca-certificate> $BMOPATH/ironic-deployment/tls/ca.crt
+   cp <path-to-tls-certificate> $BMOPATH/ironic-deployment/tls/tls.crt
```

Comment:
I'm curious how you're planning for the certificate to be generated by a user if they don't know an IP address and we don't have host names.

Reply:
If they don't know the ip address, they can't really do any setup, no ? How would they configure BMO ? What they could maybe do is generate a certificate for a domain name that also covers the 127.0.0.1 IP address with the SAN, give that as PROVISIONING_IP to ironic and inspector, and configure BMO with the domain (for example from k8s service). That could work. But if the user is deploying with the keepalived scenario, then they know the IP address.